frontend/packages/knative-plugin/src/utils/__tests__/knative-topology-utils.spec.ts

```diff
@@ -117,6 +124,19 @@ describe('knative topology utils', () => {
     const mockService = { metadata: { name: 'ser', namepspace: '' }, status: {}, spec: {} };
     expect(filterRevisionsBaseOnTrafficStatus(mockService, mockRevisions)).toBeUndefined();
   });
+
+  it('should filter out deployments created for knative resources and event sources', () => {
+    const MockResources: k8s.DeploymentKind[] = [
+      sampleEventSourceDeployments.data[0],
+      sampleKnativeDeployments.data[0],
+      sampleDeployments.data[0],
+    ];
+    const filteredResources: k8s.DeploymentKind[] = filterNonKnativeDeployments(MockResources, [
+      sampleEventSourceApiServer.data[0],
+    ]);
+    expect(filteredResources).toHaveLength(1);
+    expect(filteredResources[0].metadata?.name).toEqual('analytics-deployment');
```

Comment:
nit: Is there a chance that mock data will not have metadata.name?

Reply:
I didn't find any such event source yet. although metadata itself is optional.

Reply:
@nemesis09 i think what @sahil143 mean is these are mock data so it'll always have `metadata.name`

Reply:
This test checks of the deployments for knative service and event sources have been filtered out, leaving behind only the deployment which was not associated with either knative service or event source, i.e `analytics-deployment`